ui/src/pages/Livemap/Livemap.svelte

```diff
@@ -0,0 +1,305 @@
+<script lang="ts">
+	import { MENU_WIDE } from '@store/stores'
+	import { PLAYER, SELECTED_PLAYER } from '@store/players'
+	import Header from '@components/Header.svelte'
+	import Spinner from '@components/Spinner.svelte'
+	import Button from './components/Button.svelte'
+
+	import { onMount, tick, onDestroy } from 'svelte'
+	import { SendNUI } from '@utils/SendNUI'
+	import { get } from 'svelte/store'
+
+	import L from 'leaflet'
+	import 'leaflet/dist/leaflet.css'
+
+	let search = ''
+	let loading = false
+	let pollInterval: any;
+
+	let mapContainer: HTMLDivElement
+	let mapInitialized = false
+	let map: L.Map | null = null
+	let markers: Record<string, L.Marker> = {}
+
+	MENU_WIDE.subscribe(async (value) => {
+		if (value) {
+			await tick()
+			if (mapContainer && !mapInitialized) {
+				if (!mapContainer) {
+					console.error('Map container not found!')
+					return
+				}
+				initializeMap()
+			}
+		}
+	})
+
+	function initializeMap() {
+		mapInitialized = true
+
+		const CustomCRS = L.extend({}, L.CRS.Simple, {
+			projection: L.Projection.LonLat,
+			scale: function(zoom) {
+				return Math.pow(2, zoom);
+			},
+			zoom: function(sc) {
+				return Math.log(sc) / 0.6931471805599453;
+			},
+			distance: function(pos1, pos2) {
+				var x_difference = pos2.lng - pos1.lng;
+				var y_difference = pos2.lat - pos1.lat;
+				return Math.sqrt(x_difference * x_difference + y_difference * y_difference);
+			},
+			transformation: new L.Transformation(0.02072, 117.3, -0.0205, 172.8),
+			infinite: false
+		});
+
+		map = L.map(mapContainer, {

```

Comment:
```suggestion
		map = setupMap()
```
An extraction of this method could be something like the above, where you basically just return `L.map(.....`